src/neural_data_simulator/ephys_generator.py

```diff
@@ -181,10 +181,13 @@ def __next__(self):
         """Return the next spike event."""
         if self._current >= len(self.time_idx):
             raise StopIteration
+        waveform = (
```

Comment:
Maybe this (and below) in the `__getitem__` method could be extracted into a function so there's some documentation to what's happening (and why it might return None)?

Reply:
Thanks for the suggestion! I extracted this into a separate method and comment.